src/skspatial/objects/line_segment.py

```diff
@@ -106,14 +106,17 @@ def contains_point(self, point: array_like, **kwargs) -> bool:
 
         return math.isclose(similarity, -1, **kwargs)
 
-    def intersect_line_segment(self, other: LineSegment) -> Point:
+    def intersect_line_segment(self, other: LineSegment, tol: np.float64 = None) -> Point:
```

Comment:
This method can just pass all kwargs along to the `intersect_line` method.

```suggestion
    def intersect_line_segment(self, other: LineSegment, **kwargs) -> Point:
```